src/odemis/driver/technolution.py

```diff
@@ -1039,12 +1048,86 @@ def getCalibrationSetpoints(self, total_line_scan_time):
+            total_line_scan_time, sine_in_x=True, sine_in_y=True
+        )
+
+        # Generate scanner setpoints (note required by the ASM API, not needed for moving the descanners)
+        x_scan_setpoints, y_scan_setpoints, scan_calibration_dwell_time_ticks = \
+            self.parent._ebeam_scanner.getCalibrationSetpoints(total_line_scan_time)
+
+        # Prepare calibration data
+        calibration_data = CalibrationLoopParameters(self.rotation.value,
+                                                     0,  # Descan X offset parameter unused.
+                                                     x_descan_setpoints,
+                                                     0,  # Descan Y offset parameter unused.
+                                                     y_descan_setpoints,
+                                                     scan_calibration_dwell_time_ticks,
+                                                     self.parent._ebeam_scanner.rotation.value,
+                                                     self.parent._ebeam_scanner.getTicksScanDelay()[0],
+                                                     0.0,  # Scan X offset parameter unused.
+                                                     x_scan_setpoints,
+                                                     0.0,  # Scan Y offset parameter unused.
+                                                     y_scan_setpoints)
+
+        # Check if an acquisition is still in progress
+        if not self.parent._mppc.acq_queue.empty():
+            logging.error("There is still an unfinished acquisition in progress. "
+                          "Calibration mode cannot be started yet.")
+            return
+
+        # Stop the calibration loop if it's already running to restart it with new parameters
+        if self.parent.calibrationMode.value:
+            # Sending this command without the calibration loop being active might cause errors.
+            self.parent.asmApiPostCall("/scan/stop_calibration_loop", 204)
+
+        # Start the calibration loop with the new parameters
+        self.parent._calibrationParameters = calibration_data
+        self.parent.asmApiPostCall("/scan/start_calibration_loop",
+                                   204,
+                                   data=self.parent._calibrationParameters.to_dict())
+        time.sleep(total_line_scan_time * 10)
+        self.parent.asmApiPostCall("/scan/stop_calibration_loop", 204)
+
+        # Restore original parameters and clear calibration parameters
+        self.parent._calibrationParameters = None
+        self.parent._ebeam_scanner.dwellTime.value = init_dwellTime
+        self.physicalFlybackTime.value = init_physicalFlybackTime
+        self.scanOffset.value = init_scanOffset
+        self.scanAmplitude.value = init_scanAmplitude
+        self.parent._mppc.cellCompleteResolution.value = init_cellCompleteResolution
```

Comment:
Typically, you want to place this in a `try/finally`, so that even if the code fails the initial values are restored.